website/frontend/src/components/pages/Programs/children/Messages/Messages.tsx

```diff
@@ -1,37 +1,47 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useRef } from 'react';
 import { useSearchParams } from 'react-router-dom';
-import { GearKeyring } from '@gear-js/api';
 
-import './Messages.scss';
-import { SearchForm } from '../../blocks/SearchForm/SearchForm';
+import styles from '../../Programs.module.scss';
 
 import { getMessages } from 'services';
 import { MessageModel } from 'types/message';
+import { useAccount } from 'hooks';
 import { INITIAL_LIMIT_BY_PAGE, URL_PARAMS } from 'consts';
-import { useAccount, useChangeEffect } from 'hooks';
 import { Pagination } from 'components/Pagination/Pagination';
+import { SearchForm } from 'components/blocks/SearchForm/SearchForm';
 import { MessagesList } from 'components/blocks/MessagesList/MessagesList';
 
-export const Messages = () => {
-  const [searchParams, setSearchParams] = useSearchParams();
+const Messages = () => {
   const { account } = useAccount();
+  const isAccountLoaded = useRef(false);
+
+  const [searchParams, setSearchParams] = useSearchParams();
 
-  const page = searchParams.has(URL_PARAMS.PAGE) ? Number(searchParams.get(URL_PARAMS.PAGE)) : 1;
-  const query = searchParams.has(URL_PARAMS.QUERY) ? String(searchParams.get(URL_PARAMS.QUERY)) : '';
+  const page = Number(searchParams.get(URL_PARAMS.PAGE) ?? 1);
+  const query = searchParams.get(URL_PARAMS.QUERY) ?? '';
 
   const [messages, setMessages] = useState<MessageModel[]>([]);
   const [messagesCount, setMessagesCount] = useState(0);
 
-  useChangeEffect(() => {
-    searchParams.set(URL_PARAMS.PAGE, String(1));
-    searchParams.set(URL_PARAMS.QUERY, '');
-    setSearchParams(searchParams);
+  useEffect(() => {
+    if (isAccountLoaded.current) {
+      searchParams.set(URL_PARAMS.PAGE, String(1));
+      searchParams.set(URL_PARAMS.QUERY, '');
+      setSearchParams(searchParams);
+      setMessages([]);
+      setMessagesCount(0);
+    }
+
+    return () => {
+      isAccountLoaded.current = Boolean(account);
+    };
```

Comment:
why not just looking on `account` itself?

Reply:
Required to track logout, because at the first render account is undefined

Reply:
using useChangeEffect and looking for account is more straightforward imo

Reply:
will not work, you need to skip the first two renders with an undefined account and the first time you got an account